format/ldb/ldb.go

```diff
@@ -0,0 +1,279 @@
+package ldb
+
+// https://github.com/google/leveldb/blob/main/doc/table_format.md
+// https://github.com/google/leveldb/blob/main/doc/impl.md
+// https://github.com/google/leveldb/blob/main/doc/index.md
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"hash/crc32"
+
+	"github.com/golang/snappy"
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/pkg/bitio"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/interp"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+func init() {
+	interp.RegisterFormat(
+		format.LDB,
+		&decode.Format{
+			Description: "LevelDB Table",
```

Comment:
You can add a .md file named the same as the format and add various tips, todos and author. see maybe bson.md and bson.go how it works. It will also be used for help in the CLI and for generating formats.md with `make doc`